src/HomePage/components/MoviesList/MoviesList.tsx

```diff
@@ -4,7 +4,33 @@ import './MoviesList.css'
 import {Movie} from "../../../Mocks";
 
 export function MoviesList({ movies } : any) {
+    const [state, setState] = React.useState({
+        isActionsPopupOpen: false,
+        isCardEditable: false,
+        isCardDeletable: false,
+        cardNumber: -1
+    })
```

Comment:
I would suggest to use typescript potential and have only 2 fields in this state
```
const ModalType = 'delete' | 'edit';
//inside component
const [modal, setModal] = useState({
     videoId: null,
     operation: null
});

//...
onMovieClick={() => setModal({ videoId: movie.id, operation: 'edit' })}
onMovieDelete={() => setModal({ videoId: movie.id, operation: 'delete' })}
onMovieAdd={() => setModal({ videoId: movie.id, operation: 'add' })}
```